optimizely/optimizely_user_context.py

```diff
@@ -114,3 +147,132 @@ def as_json(self):
             'user_id': self.user_id,
             'attributes': self.get_user_attributes(),
         }
+
+    def set_forced_decision(self, decision_context, decision):
+        """
+        Sets the forced decision for a given decision context.
+
+        Args:
+            decision_context: a decision context.
+            decision: a forced decision.
+
+        Returns:
+            True if the forced decision has been set successfully.
+        """
+        if not self.client.config_manager.get_config():
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return False
+
+        with self.lock:
+            self.forced_decisions[decision_context] = decision
+
+        return True
+
+    def get_forced_decision(self, decision_context):
+        """
+        Gets the forced decision (variation key) for a given decision context.
+
+        Args:
+            decision_context: a decision context.
+
+        Returns:
+            A forced_decision or None if forced decisions are not set for the parameters.
+        """
+        if not self.client.config_manager.get_config():
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return None
+
+        forced_decision = self.find_forced_decision(decision_context)
+
+        return forced_decision if forced_decision else None
+
+    def remove_forced_decision(self, decision_context):
+        """
+        Removes the forced decision for a given decision context.
+
+        Args:
+            decision_context: a decision context.
+
+        Returns:
+            Returns: true if the forced decision has been removed successfully.
+        """
+        if not self.client.config_manager.get_config():
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return False
+
+        with self.lock:
+            if decision_context in self.forced_decisions:
+                del self.forced_decisions[decision_context]
+                return True
+
+        return False
+
+    def remove_all_forced_decisions(self):
+        """
+        Removes all forced decisions bound to this user context.
+
+        Returns:
+            True if forced decisions have been removed successfully.
+        """
+        if not self.client.config_manager.get_config():
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return False
+
+        with self.lock:
+            self.forced_decisions.clear()
+
+        return True
+
+    def find_forced_decision(self, decision_context):
+
+        with self.lock:
+            if not self.forced_decisions:
```

Comment:
forced_decisions is not the valid name, it should be forced_decisions_map

Reply:
Done, renamed.